server/Models/AdoptablePet.js

```diff
@@ -37,28 +37,33 @@ class AdoptablePet {
       if (result.rowCount) {
         const petDetailData = result.rows[0];
         const petToReturn = new this(petDetailData);
-        return petToReturn; 
+        return petToReturn;
       }
     } catch(err) {
       console.log(err);
       throw(err);
     }
   }
 
+  async save(){
+    const query = "INSERT INTO adoptable_pets (name, img_url, age, vaccination_status, adoption_story, available_for_adoption, pet_type_id) VALUES ($1, $2, $3, $4, $5, $6, $7) RETURNING id;";
+    const result = await pool.query(query, [this.name, this.imageUrl, this.age, this.vaccinationStatus,this.adoptionStory, this.availableForAdoption, this.petTypeId]);
+    return result.rows[0].id;
+  }
```

Comment:
Looks good here! Let's wrap this in a try/catch block to better handle any errors though

Reply:
Good one. Might answer bad behavior I observed client side.